test/unicode/checkstring.jl

```diff
@@ -89,11 +89,7 @@ try
         @test_throws UnicodeError Base.checkstring(UInt8[byt,0x80,0x80,0xc0])
     end
 
-    # Long encoding of 0x01
-    @test_throws UnicodeError utf8(b"\xf0\x80\x80\x80")
-    # Test ends of long encoded surrogates
-    @test_throws UnicodeError utf8(b"\xf0\x8d\xa0\x80")
-    @test_throws UnicodeError utf8(b"\xf0\x8d\xbf\xbf")
```

Comment:
Yes, the conversion stuff is a total mess right now – there's no coherent concept of what conversion means:

```jl
julia> String(UInt8[226,136,128])
"∀"

julia> String(UInt[226,136,128])
ERROR: MethodError: Cannot `convert` an object of type Array{UInt64,1} to an object of type String
This may have arisen from a call to the constructor String(...),
since type constructors fall back to convert methods.
 in String(::Array{UInt64,1}) at ./sysimg.jl:50
 in eval(::Module, ::Any) at ./boot.jl:225
 in macro expansion at ./REPL.jl:92 [inlined]
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:46

julia> String(Int32[226,136,128])
"â\u88\u80"

julia> String(UInt32[226,136,128])
"â\u88\u80"

julia> String(Char[226,136,128])
"â\u88\u80"
```

Having the meaning of calling `String` on a vector of integers depend on the details of the representation of the integers like this is a total disaster. Only `String(UInt8[...])` should work since that provides the actual data for the `String` object. Perhaps a `Vector{Char}` should be convertible to strings but it should be a convert call, not a constructor call. I've added a work item for the next round of Stringapalooza to fix this trainwreck.